app/controllers/courses_controller.rb

```diff
@@ -1,174 +1,172 @@
+      :course_credit=>cd.course.credit,
+      :open_on_latest=>(cd.course_teachership.course_details.last.semester_id==Semester::LAST.id) ? true : false ,
+      :related_cds=>cd.course_teachership.course_details.includes(:semester,:department).order("semester_id DESC"),
+      :updated_at=>cd.updated_at
+    }
+    #render "/course_content/show"
   end
-	
+
   def simulation  
-		@degree_sel=[['大學部','3'],['研究所','2'],['大學部共同','0']]
-		@dept_sel=Department.searchable.pluck(:ch_name, :id)
-		@q=CourseDetail.search({:id_in=>[0]})
+    @degree_sel=[['大學部','3'],['研究所','2'],['大學部共同','0']]
+    @dept_sel=Department.searchable.pluck(:ch_name, :id)
+    @q=CourseDetail.search({:id_in=>[0]})
+  end
+
+  def add_to_cart
+    cd_id=params[:cd_id].to_i
+    cd_to_add = CourseDetail.find(cd_id)
+    #cd_ids=JSON.parse(cookies.signed[:cd])
+    if params[:type]=="add"
+      if CourseDetail.where(:id=>cd_id).empty?
+        alt_class="warning"
+        mesg="查無此門課!"
+      else
+        unless current_user.course_details.include?(cd_to_add)
+          current_user.course_details << cd_to_add
+          alt_class="success"
+          mesg="新增成功!"
+        else
+          alt_class="info"
+          mesg="您已加入此課程!"		
+        end
+      end
+    else
+      if cd_ids.include?(cd_id)
+        cd_ids.delete(cd_id) 
+        alt_class="success"
+        mesg="刪除成功!"
+      else
+        alt_class="warning"
+        mesg="你未加入此課程!"
+      end
+    end
+    respond_to do |format|
+      format.json {
+        render :json => {:class=>alt_class, :text=>mesg}
+      }
+    end
   end
-	
-	def add_to_cart
-		cd_id=params[:cd_id].to_i
-		cookies.signed[:cd]=JSON.generate([]) if cookies.signed[:cd].nil?
-		cd_ids=JSON.parse(cookies.signed[:cd])
-		if params[:type]=="add"
-			if CourseDetail.where(:id=>cd_id).empty?
-				alt_class="warning"
-				mesg="查無此門課!"
-			else
-				if !cd_ids.include?(cd_id)
-					cd_ids.append(cd_id)
-					cd_ids=CourseDetail.select(:id).where(:id=>cd_ids).order(:time).pluck(:id)
-					alt_class="success"
-					mesg="新增成功!"
-				else
-					alt_class="info"
-					mesg="您已加入此課程!"		
-				end
-			end
-		else
-			if cd_ids.include?(cd_id)
-				cd_ids.delete(cd_id) 
-				alt_class="success"
-				mesg="刪除成功!"
-			else
-				alt_class="warning"
-				mesg="你未加入此課程!"
-			end
-		end
-		cookies.signed[:cd]={
-			:value=> JSON.generate(cd_ids),
-			:expires=> 6.months.from_now
-		}
-		respond_to do |format|
-			format.json {
-				render :json => {:class=>alt_class, :text=>mesg}
-			}
-		end
-	end
-	
-
-	
-	def show_cart
-		cookies.signed[:cd]=JSON.generate([]) if cookies.signed[:cd].nil?
-		@result={
-			:view_type=>params[:view_type],
-			:use_type=>params[:use_type],#"delete",
-			:add_to_cart=>params[:add_to_cart]=='1',
-			:courses=>CourseDetail.where(:id=>JSON.parse(cookies.signed[:cd])).map{|cd|
-				cd.to_search_result
-			}
-		}
-		render :layout=>false
-	end
-		
-  
+
+  def destroy_cart
+    current_user.user_course_detailships.destroy_all
```

Comment:
I removed this function in the next commit (Add RSpec for user's favorite courses).
Sorry for the unclear commit message. QQ